examples/with-thirdparty/app/src/main/java/com/supertokens/supertokensexample/LoginActivity.kt

```diff
@@ -0,0 +1,195 @@
+
+        githubResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+            onGithubResultReceived(it)
+        }
+    }
+
+    private fun signInWithGoogle() {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken("GOOGLE_WEB_CLIENT_ID")
+            .requestEmail()
+            .build()
+
+        val googleClient = GoogleSignIn.getClient(this, gso)
+        val signInIntent = googleClient.signInIntent
+
+        googleResultLauncher.launch(signInIntent)
+    }
+
+    private fun signInWithGitHub() {
+        val serviceConfig = AuthorizationServiceConfiguration(Uri.parse("https://github.com/login/oauth/authorize"), Uri.parse("https://github.com/login/oauth/access_token"))
+        val builder = AuthorizationRequest.Builder(
+            serviceConfig,
+            "GITHUB_CLIENT_ID",
+            ResponseTypeValues.ID_TOKEN,
+            Uri.parse("com.supertokens.supertokensexample://oauthredirect"),
+        )
+        val request = builder
+            .setScope("user")
+            .build()
+
+        val service = AuthorizationService(this)
+        val intent = service.getAuthorizationRequestIntent(request)
+
+        githubResultLauncher.launch(intent)
+    }
+
+    private fun onGithubResultReceived(it: ActivityResult) {
+        val data = it.data
+
+        if (data == null) {
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        val response = AuthorizationResponse.fromIntent(data)
+        val exception = AuthorizationException.fromIntent(data)
+
+        if (exception != null) {
+            exception.message?.let { msg -> Log.e("Login Activity", msg) }
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        if (response == null) {
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        val code = response.authorizationCode
+        val idToken = response.idToken
+        val state = response.state
+
+        val body = JSONObject()
+        body.put("thirdPartyId", "github")
+
+        if (idToken != null) {
+            val oauthTokens = JSONObject()
+            oauthTokens.put("id_token", idToken)
+            body.put("oAuthTokens", oauthTokens)
+        }
+
+        if (code != null && state != null) {
+            val redirectURIInfo = JSONObject()
+            redirectURIInfo.put("redirectURIOnProviderDashboard", "com.supertokens.supertokensexample://oauthredirect")
+
+            val redirectURIQueryParams = JSONObject()
+            redirectURIQueryParams.put("code", code)
+            redirectURIQueryParams.put("state", state)
+
+            redirectURIInfo.put("redirectURIQueryParams", redirectURIQueryParams)
+            body.put("redirectURIInfo", redirectURIInfo)
+        }
```

Comment:
we should use one of these, and not both. Since if idToken is present in the request body, then code is just ignored.

Reply:
This was more of a demonstration of how we can use either depending on how the user configures their OAuth service, but I can remove the id token bit because for the sake of this example it always uses the code